t/proxyunits.t

```diff
@@ -572,20 +719,23 @@ check_sanity($ps);
 {
     note("Test await argument:" . __LINE__);
 
+    subtest 'Await hits all 3 backends' => sub {
+        # Test Scenario
+        my $key = "/awaitbasic/a";
+        my $ps_send = "get $key\r\n";
+
+        my @be_recv = [$ps_send];
+        my %be_recv = (0 => @be_recv, 1 => @be_recv, 2 => @be_recv);
+
+        my @be_send = ["VALUE $key 0 2\r\nok\r\nEND\r\n"];
+        my %be_send = (0 => @be_send, 1 => @be_send, 2 => @be_send);
+        
+        my @ps_recv = ("VALUE $key 0 11\r\n", "hit hit hit\r\n", "END\r\n");
+        proxy_test($ps_send, \%be_recv, \%be_send, \@ps_recv);
+    };
```

Comment:
This is to show how `proxy_test` can be applies to test with multiple backends.

Reply:
must've missed that one.

guess it's fine, just feels really verbose in the tests.

Reply:
the cognitive burden hopefully can become less when it is applied in more places. :)

Reply:
trade-off between flexibility and verbosity, open for suggestions.

Reply:
Most of my point is that it can't be applied in a lot more places. Many tests are checking for order, adding delays or garbage, etc. It's necessary for testing to be flexible, and if the invocation is so large that I'm going to be copy/pasting the whole caller for each test it doesn't seem to actually help much.

For (an incomplete) example:

```
subtest 'foo' => sub {
   my $key = "/basic/a";
    basic_test("get $key\r\n",
      "get $key\r\n",
     "VALUE $key 0 2\r\nhi\r\nEND\r\n",
     ["VALUE $key 0 2\r\n", "hi\r\n", "END\r\n"]);
};
```

for "only check backend zero"; way shorter, way fewer symbols to type, etc.

then add a different function for checking all backends. don't use a function at all (or make another) if order has to be checked, etc.

For named arguments:
```
subtest 'foo' => sub {
  my $key = "/basic/a";
  basic_test(ps_send => "get $key\r\n",
    be_recv => "get $key\r\n",
    etc
  );
};

sub basic_test {
  my %a = @_;
  $a{'ps_send'};
}
```

Still much shorter than the original, a little more typing, but the arguments are named for readability.

Reply:
note syntax may not be exact but should be close :P I am groggy and don't write tons of perl anymore.

Reply:
It is not bad with `proxy_test` when testing with backend 0. 

In my mind, `proxy_test` is just for connivence and one can choose when and how to use it.

```
    subtest 'quiet flag 2' => sub {
        proxy_test(
            "ms /b/a 2 q\r\nhi\r\n", 
            {0 => ["ms /b/a 2  \r\n", "hi\r\n"]}, 
            {0 => ["EX\r\n"]}, 
            ["EX\r\n"]
        );
    };
```

Reply:
The same test looks like this in the PR, I personally prefer some verbosity for clarify (but controversial if it really adds clarify or not). 

```
    subtest 'quiet flag 2' => sub {
        my $ps_send = "ms /b/a 2 q\r\nhi\r\n";
        # backend must receive response with quiet flag replaced by a space.
        my %be_recv = (0 => ["ms /b/a 2  \r\n", "hi\r\n"]);
        my %be_send = (0 => ["EX\r\n"]);
        # proxy must return EX response from the backend.
        my @ps_recv = ("EX\r\n");
        proxy_test($ps_send, \%be_recv, \%be_send, \@ps_recv);
    };
```

Reply:
let me looking into `named arguments` and update accordingly.

I need some time to get all the gists of what you said, but I will own up what I submit and deal with future changes.

Reply:
Updated to named argument

Reply:
@dormando can this be merged?